.ci.yaml

```diff
@@ -2549,6 +2549,24 @@ targets:
       - bin/**
       - .ci.yaml
 
+  - name: Mac hello_world_macos__compile
```

Comment:
It would be nice if we could run this on both M1 and Intel Macs. I wasn't sure how to configure that in a target, so this is my best effort.

Reply:
This enables task in `x64` only.
The arm64 support is tracked here: https://github.com/flutter/flutter/issues/104034
F.Y.I. https://github.com/flutter/flutter/pull/109889 is attempting to run host only task in either arm. /cc @jmagman